conditions/conditions.go

```diff
@@ -30,13 +31,20 @@ type StatusWithConditions[S HasConditions] struct {
 }
 
 // GetStatusConditions returns all status conditions.
-func (s StatusWithConditions[S]) GetStatusConditions() []metav1.Condition {
-	return *s.Status.GetStatusConditions()
+func (s *StatusWithConditions[S]) GetStatusConditions() *[]metav1.Condition {
+	var nilS S
```

Comment:
I've been calling these variables "zero" since it's a trick to get the zero value for a type.